### PR TITLE
Add note for install of specific versions of uv

### PR DESCRIPTION
**Pull request by zanieb**
I opted not to have this in the getting started section to avoid complexity there. I figure this will move during docs restructuring soon.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,6 +140,20 @@ useful with a minimal barrier to adoption.
 
 ## Advanced Usage
 
+### Installing specific versions of uv
```
Comment (charliermarsh) [inline README.md:143]: Can we move this lower down? It seems fringe compared to other topics here.

Comment (zanieb) [reply README.md:143]: I don't think it's fringe, I feel like pinning the version is good practice and pretty normal?

Comment (zanieb) [reply README.md:143]: Do you want it under the entire advanced usage section?

Comment (zanieb) [reply README.md]: Moved down

File: README.md [hunk 2]
```diff
@@ -62,6 +62,8 @@ pipx install uv
 brew install uv
 ```
 
+See the [advanced installation](#advanced-installation) documentation for more details.
```
Comment (charliermarsh) [inline README.md:65]: Can we remove this?

File: README.md [hunk 3]
```diff
@@ -481,6 +483,36 @@ If a direct path to the certificate is required (e.g., in CI), set the `SSL_CERT
 variable to the path of the certificate bundle, to instruct uv to use that file instead of the
 system's trust store.
 
+## Advanced installation
+
+### Installing a specific version
+
+uv's standalone installer can also be pinned to a specific version, for example, to install version `0.2.6`:
```
Comment (charliermarsh) [inline README.md:490]: For consistency:

```
uv's standalone installer can also be pinned to a specific version. For example,
```

File: README.md [hunk 4]
```diff
@@ -481,6 +483,36 @@ If a direct path to the certificate is required (e.g., in CI), set the `SSL_CERT
...
+
+Or, specifiy a constraint range to allow upgrades without breaking changes:
+
+```shell
+pip install 'uv>=0.2,<0.3'
+```
+
+Note the standalone installer does not support version ranges.
+
+If using other package managers for installation of uv, consult their documentation for requesting specific versions.
```
Comment (charliermarsh) [inline README.md:514]: I would omit this.

**Review by charliermarsh — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (zanieb) [reply README.md:65]: I really do not want people to have to scroll to the bottom of the documentation to discover how to pin the version with the standalone installer. Can we compromise here? I'm happy to rephrase it.

Comment (charliermarsh) [reply README.md:65]: But this doesn't actually tell people how to pin versions with the standalone installer. Is it more useful than having to Command + F for "install"? (If it's that important, should it be up here in the preamble?)

Comment (charliermarsh) [reply README.md:65]: E.g., could the whole change just be a line like:

```
# Or a specific version.
curl -LsSf https://astral.sh/uv/0.2.6/install.sh | sh
```

In the above code block? Isn't that more economical than adding this to the real estate of the Getting Started section?

Comment (zanieb) [reply README.md:65]: > But this doesn't actually tell people how to pin versions with the standalone installer.

That's what I was getting at when I said I'd rephrase.

>  Isn't that more economical than adding this to the real estate of the Getting Started section?

Yeah. So that was my first thought, but there are two small problems

1. I don't want the getting started to suggest installing an out of date version. We can resolve this by adding the README to rooster's target files though so it's always the latest in the example.
2. It's a little awkward given that we split between unix and windows, `# Or, a specific version on MacOS and Linux`?

I'd prefer this honestly just not sure how to phrase it since it doesn't quite fit into the style of the rest of the code block.

Comment (zanieb) [reply README.md:65]: Okay: https://github.com/astral-sh/uv/pull/4105

**Comment by zanieb** [comment]
Using #4105 instead
